### DISCUSSION TITLE
Saving Session next_joined and current_joined attributes

### DISCUSSION THREAD

**Discussion by Gary-Manuel** [BottlecapDave/HomeAssistant-OctopusEnergy#555] [General]
Hi Dave. First of all, thanks for the latest update v9.0.0 with the saving session auto opt in. Just what I’ve been waiting for.
One thing I’m not sure about and that is the role played by the newly added current_joined attributes.  Previously, the next_joined attributes were populated from the time of joining, up to the session start time, then they disappeared. Is this still the case and are the current_joined attributes just populated between the start and end times?

**Comment by BottlecapDave** [comment]
Omission on my part. These attributes were present in before in the old carnation of the sensor, but I've updated the [docs](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/_docs/entities/octoplus.md#saving-sessions) with some explanations on those attributes. But long story short, yes the next attributes work how they did before. The current attributes will become available when the session is active.

**Reply by Gary-Manuel** [reply]
That’s great. Thanks for the quick response.

Bit of proof-reading feedback about the docs for you. You have the “Type” of both event_end attributes as “boolean,” rather than “datetime”. Hope you don’t mind me pointing this out.

**Reply by BottlecapDave** [reply]
No problem and no worry pointing out errors. I've updated the docs.
